src/internal/xml-parser.ts

```diff
@@ -0,0 +1,87 @@
+import type * as http from 'node:http'
+
+import { XMLParser } from 'fast-xml-parser'
+
+import * as errors from '../errors.ts'
+import { parseXml } from './helper.ts'
+import { readAsString } from './response.ts'
+
+// parse XML response for bucket region
+export function parseBucketRegion(xml: string): string {
+  // return region information
+  return parseXml(xml).LocationConstraint
+}
+
+const fxp = new XMLParser()
+
+// Parse XML and return information as Javascript types
+// parse error XML response
+export function parseError(xml: string, headerInfo: Record<string, any>) {
```

Comment:
I think this should be `never` or `unknown` instead of `any`. But for now I guess `any` will work.

Reply:
yes, `unknown` would be better